Utils/FaviconFetcher.swift

```diff
@@ -93,73 +94,84 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
         return deferred
     }
 
+    private func iconTypeForLink(rel:String) -> IconType {
+        var iconType = IconType.Guess
+        switch (rel) {
+        case "shortcut icon":
+            iconType = .Icon
+        case "icon":
+            iconType = .Icon
+        case "apple-touch-icon":
+            iconType = .AppleIcon
+        case "apple-touch-icon-precomposed":
+            iconType = .AppleIconPrecomposed
+        default:
+            iconType = IconType.Guess
+        }
+        return iconType
+    }
+
     // Loads and parses an html document and tries to find any known favicon-type tags for the page
     private func parseHTMLForFavicons(url: NSURL) -> Deferred<Maybe<[Favicon]>> {
+
         return fetchDataForURL(url).bind({ result -> Deferred<Maybe<[Favicon]>> in
             var icons = [Favicon]()
+            guard let data = result.successValue where result.isSuccess else {
+                return deferMaybe(icons)
+            }
 
-            if let data = result.successValue where result.isSuccess,
-               let element = RXMLElement(fromHTMLData: data) where element.isValid {
-                var reloadUrl: NSURL? = nil
-                element.iterate("head.meta") { meta in
-                    if let refresh = meta.attribute("http-equiv") where refresh == "Refresh",
-                        let content = meta.attribute("content"),
-                        let index = content.rangeOfString("URL="),
-                        let url = NSURL(string: content.substringFromIndex(index.startIndex.advancedBy(4))) {
-                            reloadUrl = url
-                    }
-                }
-
-                if let url = reloadUrl {
-                    return self.parseHTMLForFavicons(url)
-                }
+            let element = SWXMLHash.lazy(data)
```

Comment:
Made sense not to create a parse tree for the entire page when we only need the head. Lazy loading should help with that.